src/api/l_physics_world.c

```diff
@@ -40,25 +40,25 @@ static void raycastCallback(Shape* shape, float x, float y, float z, float nx, f
 
 static int l_lovrWorldNewCollider(lua_State* L) {
   World* world = luax_checktype(L, 1, World);
-  float x = luax_optfloat(L, 2, 0.f);
-  float y = luax_optfloat(L, 3, 0.f);
-  float z = luax_optfloat(L, 4, 0.f);
-  Collider* collider = lovrColliderCreate(world, x, y, z);
+  float position[4];
+  luax_readvec3(L, 2, position, NULL);
+  Collider* collider = lovrColliderCreate(world, position[0], position[1], position[2]);
   luax_pushtype(L, Collider, collider);
   lovrRelease(Collider, collider);
   return 1;
 }
 
 static int l_lovrWorldNewBoxCollider(lua_State* L) {
   World* world = luax_checktype(L, 1, World);
-  float x = luax_optfloat(L, 2, 0.f);
-  float y = luax_optfloat(L, 3, 0.f);
-  float z = luax_optfloat(L, 4, 0.f);
-  float sx = luax_optfloat(L, 5, 1.f);
-  float sy = luax_optfloat(L, 6, sx);
-  float sz = luax_optfloat(L, 7, sx);
-  Collider* collider = lovrColliderCreate(world, x, y, z);
-  BoxShape* shape = lovrBoxShapeCreate(sx, sy, sz);
+  float position[4], size[4];
+  int index = luax_readvec3(L, 2, position, NULL);
+  luax_readvec3(L, index, size, NULL);
+  for (int i = 0; i < 3; i++) { // defaults
+    if (size[i] == 0.f)
+      size[i] = i == 0 ? 1.f : size[0];
+  }
```

Comment:
There is a function `luax_readscale` that could be used here, it fills in defaults like this.  The behavior is _slightly_ different since it applies the default if the argument is missing instead of zero, but this should be okay.

I don't mind applying this change myself after merging, just let me know.

Reply:
Fixed.